x-pack/test/observability_ai_assistant_api_integration/common/create_llm_proxy.ts

```diff
@@ -355,3 +384,31 @@ async function getRequestBody(request: http.IncomingMessage): Promise<ChatComple
 function sseEvent(chunk: unknown) {
   return `data: ${JSON.stringify(chunk)}\n\n`;
 }
+
+function extractDocumentsFromMessage(content: string, log: ToolingLog): any[] {
+  const startIndex = content.indexOf('Documents:\n');
+
+  if (startIndex === -1) {
+    log.error('Documents section not found in content.');
+    return [];
+  }
+
+  const documentSection = content.substring(startIndex + 'Documents:\n'.length).trim();
+
+  const startBracketIndex = documentSection.indexOf('[');
+  const endBracketIndex = documentSection.lastIndexOf(']');
+
+  if (startBracketIndex === -1 || endBracketIndex === -1 || endBracketIndex < startBracketIndex) {
+    log.error('Malformed documents array in content.');
+    return [];
+  }
+
+  const documentsString = documentSection.substring(startBracketIndex, endBracketIndex + 1).trim();
```

Comment:
Perhaps we can simplify this with a regex to match the JSON

```suggestion
  const documentsString = str.match(/\[\s*{[\s\S]*}\s*\]/);
```

Reply:
Actually, perhaps this regex works better

```suggestion
  const documentsString = str.match(/\{[\s\S]*?\}/g);
```

Reply:
... and can you please update `interceptSelectRelevantFieldsToolChoice` to something like this:

```ts
      arguments: (requestBody) => {
        const messageWithFieldIds = last(requestBody.messages);
        const matches = (messageWithFieldIds?.content as string).match(/\{[\s\S]*?\}/g)!;
        relevantFields = matches
          .slice(from, to)
          .map((jsonStr) => JSON.parse(jsonStr) as RelevantField);

        return JSON.stringify({ fieldIds: relevantFields.map(({ id }) => id) });
      },
```

Reply:
Yes, for sure.. it's updated